app/src/ui/theme-selector/ThemeSelector.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+
+import { Button } from "ui/button/Button";
+import { Icon } from "ui/icon/Icon";
+import { useLocalStorage } from "hooks/useLocalStorage/useLocalStorage";
+
+import { ThemeSelectorProps, ThemeState } from "./ThemeSelector.types";
+
+export const ThemeSelector: React.FC<ThemeSelectorProps> = () => {
+  const localStorage = useLocalStorage();
+  const [userTheme, setUserTheme] = useState<ThemeState>({ hasChanged: false, theme: "dark" });
+
+  useEffect(() => {
+    const localTheme = localStorage.get<ThemeState["theme"] | undefined>("theme");
+
+    if (!localTheme) {
+      localStorage.set("theme", userTheme.theme);
+      document.body.dataset.theme = userTheme.theme;
+
+      return;
+    }
+
+    document.body.dataset.theme = localTheme;
+    setUserTheme({ ...userTheme, theme: localTheme });
+  }, []);
+
+  useEffect(() => {
+    if (userTheme.hasChanged) {
```

Comment:
The `hasChanged` logic is redundant. 

`useEffect` will know if the properties in the array arg have changed.